Fix Chip classes to respect iconTheme constructor param. fixes #111828

timmaffett opened the pull request:
This PR fixes #111828
The Chip classes (`Chip`,`ActionChip`,`ChoiceChip`,`FilterChip` and `InputChip`) are corrected to pass their `iconTheme` parameter on to the `RawChip` constructor.
It adds tests to the test files for each of the `xxx_chip_test.dart` files to verify that the `iconTheme` parameter is respected (which would have caught the error that caused #111828 )
It changes `chip_theme_test.dart` to create chip's using `Chip()` (instead of `RawChip()`), and adds tests of the correct shadowing of `iconTheme`'s from `ChipThemeData`.  `Chip()` is used instead of `RawChip()` as this is a chance to test both `Chip()` and `RawChip()`.  Previosuly  `RawChip() was used this would miss the error that caused #111828 .

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

timmaffett committed:
fix Chip classes to pass iconTheme on to RawChip

timmaffett committed:
rebase to origin master

timmaffett committed:
added test to ensure iconTheme constructor param is respected

timmaffett committed:
added test to ensure iconTheme constructor param is respected

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

timmaffett committed:
dart format fixes

timmaffett committed:
dart format fixes

timmaffett committed:
Merge branch 'fix_chip_icontheme' of https://github.com/timmaffett/flutter into fix_chip_icontheme

werainkhatri commented:
thanks you for the contribution!

looks like there are merge conflicts. try rebasing from master and resolve them.

also, analysis checks are failing. check the logs here on github or view them locally by running `../../bin/flutter analyze` in the package/flutter directory.

timmaffett committed:
Merge branch 'master' into fix_chip_icontheme

timmaffett committed:
flutter analyze fixes to remove unnecessary_const

timmaffett committed:
flutter analyze fixes to remove unnecessary_const

timmaffett committed:
dart formatting

timmaffett commented:
@werainkhatri  i addressed the issues and it is now green

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

timmaffett commented:
@darrenaustin I might mention that the testing code that I added for each of the Chip classes is a little different then the test added to `ChoiceChip` by https://github.com/flutter/flutter/pull/110908
I am actually testing that the `iconTheme` makes it all the way to the avatar icon being used, instead of just checking that it makes it to the `rawChip`'s `iconTheme` property.

Also I believe that the changes to `chip_theme_test.dart` to use `Chip` instead of just `RawChip` is a better approach because this still test's `RawChip`, as `Chip` uses that, but it also tests that all of the constructor parameters are passed through to `RawChip`.
I also added test here to ensure that the `iconTheme` within `ChipThemeData` is used when present, and also that if an `iconTheme` is passed to the `RawChip` constructor that it overrides `iconTheme` defaults set at 'higher' levels within `ChipThemeData`.

werainkhatri commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -36,6 +36,15 @@ DefaultTextStyle getLabelStyle(WidgetTester tester) {
   );
 }
 
+IconTheme getIconTheme(WidgetTester tester) {
+  return tester.widget<IconTheme>(
+    find.descendant(
+      of: find.byType(RawChip),
+      matching: find.byType(IconTheme),
+    ),
+  );
+}
```
same as [`chip_test.dart`](https://github.com/flutter/flutter/blob/64f84f6a47d276e4d79cb10c203adbddeeeae336/packages/flutter/test/material/chip_test.dart#L37).
```suggestion
IconThemeData getIconData(WidgetTester tester) {
  final IconTheme iconTheme = tester.firstWidget(
    find.descendant(
      of: find.byType(RawChip),
      matching: find.byType(IconTheme),
    ),
  );
  return iconTheme.data;
}
```

timmaffett replied:
fixed

werainkhatri reviewed (changes requested):
thanks for uncovering other un-tested parts of the code.

overall comments:
1. nit pick: fix minor space and comma issues (commented on a few examples)
2. IMO only `chip_test` needs to test `RawChip`'s behavior. other chips need not do it. (thoughts @darrenaustin?)
3.
> I also added test here to ensure that the iconTheme within ChipThemeData is used when present, and also that if an iconTheme is passed to the RawChip constructor that it overrides iconTheme defaults set at 'higher' levels within ChipThemeData.

this is not tested (good catch!). but there exists 3 tests ("Chip uses ThemeData chip theme", "Chip uses ChipTheme" and "Chip uses constructor parameters") which do the same thing but for all other parameters. IMO, `iconTheme` should just be added in all of them instead of a separate test specifically for it.

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

timmaffett commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -271,6 +428,64 @@ void main() {
     expect(getLabelStyle(tester).style.fontSize, 32);
   });
 
+testWidgets('Chip uses constructor parameters with Avatar and IconThemeData', (WidgetTester tester) async {
```
merged into existing test

timmaffett replied:
merged with existing test

timmaffett commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -161,11 +173,96 @@ void main() {
+  testWidgets('Chip with Avatar Icon uses ClipThemeData icon theme', (WidgetTester tester) async {
+    const IconThemeData customIconTheme = IconThemeData(color: Color(0xff112233));
+    const ChipThemeData chipTheme = ChipThemeData(
+      backgroundColor: Color(0xff112233),
+      elevation: 4,
+      padding: EdgeInsets.all(50),
+      labelPadding: EdgeInsets.all(25),
+      shape: RoundedRectangleBorder(),
+      labelStyle: TextStyle(fontSize: 32),
+      iconTheme: customIconTheme,
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData.light().copyWith(
+          chipTheme: chipTheme,
+        ),
+        home: const Directionality(
+          textDirection: TextDirection.ltr,
+          child: Material(
+            child: Center(
+              child: Chip(
+                label: SizedBox(width: 100, height: 100),
+                avatar: Icon(Icons.ac_unit),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final RenderBox materialBox = getMaterialBox(tester);
+    expect(materialBox, paints..rect(color: chipTheme.backgroundColor));
+    expect(getMaterial(tester).elevation, chipTheme.elevation);
+    expect(tester.getSize(find.byType(RawChip)), const Size(400, 250)); // icon + label + padding + labelPadding
+    expect(getMaterial(tester).shape, chipTheme.shape);
+    expect(getLabelStyle(tester).style.fontSize, 32);
+    expect(tester.widget<IconTheme>(find.widgetWithIcon(IconTheme,Icons.ac_unit).first).data.color, customIconTheme.color);
+    expect(getIconTheme(tester).data.color, customIconTheme.color );
+  });
```
merged with existing test

werainkhatri replied:
that's a good point. i feel like a properly named function with optional defaulted arguments would do the trick quite nicely. anyway we are **unit** testing `ChipTheme` and not `RawChip`, so verifying `Material` seems overkill to me. lemme know (by re-requesting a review) when it's done. thanks!

timmaffett commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -161,11 +173,96 @@ void main() {
+    expect(tester.getSize(find.byType(RawChip)), const Size(400, 250)); // icon + label + padding + labelPadding
+    expect(getMaterial(tester).shape, chipTheme.shape);
+    expect(getLabelStyle(tester).style.fontSize,14);
+    expect(tester.widget<IconTheme>(find.widgetWithIcon(IconTheme,Icons.ac_unit).first).data.color, customIconTheme.color);
```
fixed

timmaffett replied:
fixed

timmaffett commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -161,11 +173,96 @@ void main() {
+    expect(getMaterial(tester).shape, chipTheme.shape);
+    expect(getLabelStyle(tester).style.fontSize,14);
+    expect(tester.widget<IconTheme>(find.widgetWithIcon(IconTheme,Icons.ac_unit).first).data.color, customIconTheme.color);
+    expect(getIconTheme(tester).data.color, customIconTheme.color );
```
fixed

timmaffett replied:
fixed

timmaffett commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -161,11 +173,96 @@ void main() {
+        theme: ThemeData.light().copyWith(
+          chipTheme: chipTheme,
+        ),
+        home: const Directionality(
```
I removed `Directionality` from the all of the test as you have requested.

timmaffett replied:
removed the `Directionality` test as requested.  (I had just copied from existing test, but I have now removed all the of uses in this file).

timmaffett commented on packages/flutter/test/material/action_chip_test.dart:
```diff
@@ -58,4 +58,23 @@ void main() {
+    );
+
+    expect(tester.widget<IconTheme>(find.widgetWithIcon(IconTheme,Icons.cabin).first).data.color, const Color(0xff112233));
+    expect(tester.takeException(), null);
```
Each of the xxxChip classes (`Chip`,`ActionChip`,`ChoiceChip`,`FilterChip` and `InputChip`), which all use `RawChip()` internally, were *not* passing on their `iconTheme` parameters to `RawChip()`, so I have fixed the code within each class and then added these similar tests to the test files for each of these classes to verify that the iconTheme is passed to `RawChip()`.

timmaffett replied:
simplified to only test if `RawChip` gets `iconTheme`, did same for `ActionChip`, `ChoiceChip`, `FilterChip`, `InputChip`

werainkhatri replied:
i meant why are you verifying if `tester.takeExeption` is null?

timmaffett committed:
suggest changes from code reviews, simplified tests for ActionChip,ChoiceChip,FilterChip,and InputChip

timmaffett committed:
suggest changes from code reviews, simplified tests for ActionChip,ChoiceChip,FilterChip,and InputChip

timmaffett committed:
Merge branch 'fix_chip_icontheme' of https://github.com/timmaffett/flutter into fix_chip_icontheme

timmaffett commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -213,12 +309,74 @@ void main() {
+    expect(getLabelStyle(tester).style.fontSize, 32);
+    expect(tester.widget<IconTheme>(find.widgetWithIcon(IconTheme,Icons.ac_unit).first).data.color, customIconTheme.color);
+    expect(getIconTheme(tester).data.color,customIconTheme.color,reason:'Chip Icon Theme does not reflect custom iconTheme color.');
+  });
```
fixed

timmaffett committed:
removed extra newline

timmaffett committed:
removed extra newline

timmaffett commented on packages/flutter/test/material/choice_chip_test.dart:
```diff
@@ -132,4 +151,5 @@ void main() {
     final RawChip rawChip = tester.widget(find.byType(RawChip));
     expect(rawChip.iconTheme, iconTheme);
   });
+
```
fixed

timmaffett commented on packages/flutter/test/material/choice_chip_test.dart:
```diff
@@ -121,6 +121,25 @@ void main() {
+      ),
+    );
+
+    expect(tester.widget<IconTheme>(find.widgetWithIcon(IconTheme,Icons.cabin).first).data.color, const Color(0xffeeaadd));
```
fixed

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

werainkhatri commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -161,11 +174,54 @@ void main() {
+    expect(tester.getSize(find.byType(RawChip)), const Size(400, 250)); // icon + label + padding + labelPadding
     expect(getMaterial(tester).shape, chipTheme.shape);
     expect(getLabelStyle(tester).style.fontSize, 32);
+    expect(tester.widget<IconTheme>(find.widgetWithIcon(IconTheme,Icons.ac_unit).first).data.color, customIconTheme.color);
+    expect(getIconTheme(tester).color, customIconTheme.color);
```
what's the difference b/w these two? aren't they just two different ways to find the same `IconTheme` widget? the second one will suffice.

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

timmaffett committed:
Merge branch 'master' into fix_chip_icontheme

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

werainkhatri reviewed (commented):
very sorry for the delayed review. life just got busy.

this LGTM module some minor nits (will commit them myself via github) and the unresolved conversation above.

also, there was a merge conflict, which i resolved via github. this is a pretty cool website.

werainkhatri committed:
fixes minor style issues

werainkhatri committed:
fixes minor style issues

werainkhatri committed:
Merge branch 'master' into fix_chip_icontheme

werainkhatri commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -200,14 +256,67 @@ void main() {
+      labelPadding: EdgeInsets.all(25),
+      labelStyle: TextStyle(fontSize: 32),
+      shape: RoundedRectangleBorder(),
+      iconTheme: shadowedClipThemeIconTheme,
```
well, i messed up a bit (not the first time while using GH to write code). fixing...

```suggestion
      iconTheme: shadowedChipThemeIconTheme,
```

werainkhatri committed:
fix variable name change

werainkhatri committed:
fix variable name change

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

timmaffett commented:
> very sorry for the delayed review. life just got busy.
> 
> this LGTM module some minor nits (will commit them myself via github) and the unresolved conversation above.
> 
> also, there was a merge conflict, which i resolved via github. this is a pretty cool website.

Thank you for resolving the simple nits and resolving the merge conflict.

timmaffett committed:
remove redundant material checks

timmaffett committed:
remove redundant material checks

timmaffett committed:
test only that custom iconTheme made it to rawChip via getIconTheme

timmaffett committed:
test only that custom iconTheme made it to rawChip via getIconTheme

timmaffett commented:
Hi @werainkhatri - ok, I think I have removed the redundant material test you are talking about.   

I now just check to verify that the `customIconTheme` made it to the rawChip's `iconTheme`.

werainkhatri commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -167,11 +180,54 @@ void main() {
+      ),
+    );
+
+    final RenderBox materialBox = getMaterialBox(tester);
+    expect(materialBox, paints..rect(color: chipTheme.backgroundColor));
+    expect(getMaterial(tester).elevation, chipTheme.elevation);
+    expect(tester.getSize(find.byType(RawChip)), const Size(400, 250)); // icon + label + padding + labelPadding
+    expect(getMaterial(tester).shape, chipTheme.shape);
+    expect(getLabelStyle(tester).style.fontSize, 14);
+    expect(tester.widget<IconTheme>(find.widgetWithIcon(IconTheme, Icons.ac_unit).first).data.color, customIconTheme.color);
+    expect(getIconTheme(tester).color, customIconTheme.color);
```
```suggestion
    expect(tester.getSize(find.byType(RawChip)), const Size(400, 250)); // icon + label + padding + labelPadding
    expect(getIconTheme(tester).color, customIconTheme.color);
```

werainkhatri commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -269,12 +375,8 @@ void main() {
-    expect(getMaterial(tester).elevation, elevation);
-    expect(getMaterial(tester).shape, shape);
-    expect(getLabelStyle(tester).style.fontSize, 32);
+    expect(tester.getSize(find.byType(RawChip)), const Size(400, 250)); // icon + label + padding + labelPadding
+    expect(getIconTheme(tester), customIconTheme, reason:'Chip Icon Theme does not reflect custom iconTheme.');
```
```suggestion
    final RenderBox materialBox = getMaterialBox(tester);
    expect(materialBox, paints..circle(color: backgroundColor));
    expect(tester.getSize(find.byType(RawChip)), const Size(400, 250)); // icon + label + padding + labelPadding
    expect(getMaterial(tester).elevation, elevation);
    expect(getMaterial(tester).shape, shape);
    expect(getLabelStyle(tester).style.fontSize, 32);
    expect(getIconTheme(tester), customIconTheme, reason:'Chip Icon Theme does not reflect custom iconTheme.');
```

werainkhatri reviewed (approved):
![literal stamp](https://user-images.githubusercontent.com/44755140/235419026-6238e339-0d68-455d-a8d6-b07948f534fc.jpg)

let's not alter existing tests. i intended to suggest doing it on new tests only. i'll do that via GH.

sorry for such delay and thank you for your patience. i've requested for a secondary review.

werainkhatri committed:
Apply suggestions from code review

werainkhatri committed:
Apply suggestions from code review

TahaTesser commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -105,6 +117,7 @@ void main() {
       brightness: Brightness.dark,
       elevation: 5,
       pressElevation: 6,
+      iconTheme: customIconTheme,
```
Here we'll need to increment the previous color  (`checkmarkColor: Color(0xfffffff7)` 
So this becomes:
`iconTheme: IconThemeData(color: Color(0xfffffff8)),`

This will expect a hash code when matching `IconThemeData`. We need slightly updated the test to use `equalsIgnoringHashCodes` on the list of strings.

```dart
    expect(
      description,
      equalsIgnoringHashCodes(<String>[
        'backgroundColor: Color(0xfffffff0)',
        'deleteIconColor: Color(0xfffffff1)',
        'disabledColor: Color(0xfffffff2)',
        'selectedColor: Color(0xfffffff3)',
        'secondarySelectedColor: Color(0xfffffff4)',
        'shadowColor: Color(0xfffffff5)',
        'surfaceTintColor: Color(0xfffffff8)',
        'selectedShadowColor: Color(0xfffffff6)',
        'showCheckmark: true',
        'checkMarkColor: Color(0xfffffff7)',
        'labelPadding: EdgeInsets.all(1.0)',
        'padding: EdgeInsets.all(2.0)',
        'side: BorderSide(width: 10.0)',
        'shape: RoundedRectangleBorder(BorderSide(width: 0.0, style: none), BorderRadius.zero)',
        'labelStyle: TextStyle(inherit: true, size: 10.0)',
        'secondaryLabelStyle: TextStyle(inherit: true, size: 20.0)',
        'brightness: dark',
        'elevation: 5.0',
        'pressElevation: 6.0',
        'iconTheme: IconThemeData#00000(color: Color(0xfffffff8))',
      ]),
    );
```

TahaTesser replied:
@timmaffett Please add `equalsIgnoringHashCodes` from above.

The test is failing because of this. This ignores the hashcode generated by `IconThemeData`.

TahaTesser commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -219,47 +265,101 @@ void main() {
+    expect(tester.getSize(find.byType(RawChip)), const Size(400, 250)); // icon + label + padding + labelPadding
+    expect(getIconTheme(tester), customIconTheme, reason:'Chip Icon Theme does not reflect custom iconTheme.');
+  });
+
```
Looks like this test has a confusing title and bloated themes.  We're only if the theme is overridden by `Chip.iconTheme`

Please replace this with the following:

```dart
  testWidgets('ChipTheme.iconTheme overrides local ChipTheme', (WidgetTester tester) async {
    const IconThemeData iconTheme = IconThemeData(color: Color(0xffbbee12));
    await tester.pumpWidget(
      const MaterialApp(
        home: ChipTheme(
          data: ChipThemeData(
            iconTheme: IconThemeData(color: Color(0xffbbeeff)),
          ),
          child: Material(
            child: Center(
              child: Chip(
                label: SizedBox(width: 100, height: 100),
                avatar: Icon(Icons.ac_unit),
                iconTheme: iconTheme,
              ),
            ),
          ),
        ),
      ),
    );

    expect(getIconTheme(tester), iconTheme);
  });
```

TahaTesser commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -271,10 +371,11 @@ void main() {
     expect(getMaterial(tester).elevation, elevation);
     expect(getMaterial(tester).shape, shape);
     expect(getLabelStyle(tester).style.fontSize, 32);
+    expect(getIconTheme(tester), customIconTheme, reason:'Chip Icon Theme does not reflect custom iconTheme.');
```
This is not necessary. 
```suggestion
    expect(getIconTheme(tester), customIconTheme);
```

TahaTesser commented on packages/flutter/test/material/chip_theme_test.dart:
```diff
@@ -167,11 +180,48 @@ void main() {
+    const IconThemeData customIconTheme = IconThemeData( size: 30, color: Color(0xff332211) );
+
+    await tester.pumpWidget(
+      MaterialApp(
```
Please replace this with:
```dart
  testWidgets('Chip.iconTheme overrides from ChipTheme.iconTheme from ThemeData', (WidgetTester tester) async {
    const IconThemeData iconTheme = IconThemeData(color: Color(0xff332211));
    await tester.pumpWidget(
      MaterialApp(
        theme: ThemeData(
          chipTheme: const ChipThemeData(
            iconTheme: IconThemeData(color: Color(0xff112233)),
          ),
        ),
        home: const Material(
          child: Center(
            child: Chip(
              label: SizedBox(width: 100, height: 100),
              avatar: Icon(Icons.ac_unit),
              iconTheme: iconTheme,
            ),
          ),
        ),
      ),
    );

    expect(getIconTheme(tester), iconTheme);
  });
```

TahaTesser reviewed (changes requested):
This needs a few changes in order to land.

TahaTesser reviewed (changes requested):
This needs a few changes in order to land.

timmaffett committed:
made requested changes to tests

timmaffett committed:
made requested changes to tests

timmaffett commented:
@TahaTesser I have made the requested changes.

timmaffett committed:
Merge branch 'master' into fix_chip_icontheme

TahaTesser commented:
> @TahaTesser I have made the requested changes. Please take a look when you have a chance.

Looks like `equalsIgnoringHashCodes` is missing. Please https://github.com/flutter/flutter/pull/112001#discussion_r1211535358

timmaffett commented:
The test check failures do not seem to be related to this pr?
Updating the branch and giving it another go.

timmaffett committed:
Merge branch 'master' into fix_chip_icontheme

timmaffett committed:
add equalsIgnoringHashCode for iconThemeData comparison

timmaffett committed:
add equalsIgnoringHashCode for iconThemeData comparison

timmaffett committed:
Merge branch 'master' into fix_chip_icontheme

timmaffett committed:
Merge branch 'master' into fix_chip_icontheme

timmaffett committed:
fix expect values

timmaffett committed:
fix expect values

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

timmaffett commented:
@timmaffett  The conflicts should now be resolved.

TahaTesser commented:
@timmaffett 
Thanks so much for all the work. Looks like this is still out of sync, still have conflicts?


![Screenshot 2023-06-30 at 13 06 29](https://github.com/flutter/flutter/assets/48603081/07d1ef97-de0e-4008-b20c-c9052db179e7)

(since this PR is polluted and 9 months old, you might file a new clean PR with all conflicts resolved)

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

timmaffett committed:
Merge branch 'fix_chip_icontheme' of https://github.com/timmaffett/flutter into fix_chip_icontheme

timmaffett committed:
Merge branch 'master' into fix_chip_icontheme

TahaTesser commented:
@timmaffett 
Looks like you have analyzer issues
```dart
║    info • Use 'const' with the constructor to improve performance • packages/flutter/test/material/chip_theme_test.dart:179:25 • prefer_const_constructors
║    info • Use 'const' with the constructor to improve performance • packages/flutter/test/material/chip_theme_test.dart:387:31 • prefer_const_constructors
║    info • Use 'const' with the constructor to improve performance • packages/flutter/test/material/chip_theme_test.dart:390:32 • prefer_const_constructors
║    info • Use 'const' with the constructor to improve performance • packages/flutter/test/material/chip_theme_test.dart:391:37 • prefer_const_constructors
║    info • Use 'const' with the constructor to improve performance • packages/flutter/test/material/chip_theme_test.dart:392:35 • prefer_const_constructors
║    info • Use 'const' with the constructor to improve performance • packages/flutter/test/material/chip_theme_test.dart:394:30 • prefer_const_constructors
```

timmaffett committed:
add some const for new analyzer settings

timmaffett committed:
Merge branch 'flutter:master' into fix_chip_icontheme

timmaffett commented:
@TahaTesser  I added those const the analyzer was complaining about.

TahaTesser commented:
@timmaffett 
As I mentioned [earlier](https://github.com/flutter/flutter/pull/112001#issuecomment-1614436438) this PR is very polluted, it would be better to file a new PR for this with clean changes. I still see `ci.yaml` test is failing and can't rebase this PR from UI. 

![Screenshot 2023-07-12 at 11 38 18](https://github.com/flutter/flutter/assets/48603081/dfe0a1a1-325c-4743-8e87-d924b739b86f)


I recommend closing this PR and filing a clean new PR and we can land it much sooner.

Hixie commented:
@timmaffett Thanks for your contribution. Per the comments above, it would be helpful if you could file a new PR with a clean commit with the changes, for easier review. I'm going to close this one for now since we can't review it in its current state. Thanks!
